### PR TITLE
Disable setting fields in case we use environment variables

### PR DESCRIPTION
**Pull request by nupplaphil**
As described in https://github.com/friendica/docker/issues/224 , it is possible to overwrite site settings even in case they are directly set through environment variables.

But in this case, we only save the variable but nevertheless print the environment variable again.

With this change, all fields, set by environment variables, are now disabled dynamically.

### CODE REVIEW

File: src/Module/Admin/Logs/Settings.php [hunk 1]
```diff
@@ -82,9 +88,9 @@ protected function content(array $request = []): string
 			'$clear' => DI::l10n()->t('Clear'),
 			'$logname' => DI::config()->get('system', 'logfile'),
 			// see /help/smarty3-templates#1_1 on any Friendica node
-			'$debugging' => ['debugging', DI::l10n()->t("Enable Debugging"), DI::config()->get('system', 'debugging'), ""],
-			'$logfile' => ['logfile', DI::l10n()->t("Log file"), DI::config()->get('system', 'logfile'), DI::l10n()->t("Must be writable by web server. Relative to your Friendica top-level directory.")],
-			'$loglevel' => ['loglevel', DI::l10n()->t("Log level"), DI::config()->get('system', 'loglevel'), "", $log_choices],
+			'$debugging' => ['debugging', DI::l10n()->t("Enable Debugging"), DI::config()->get('system', 'debugging'), "", DI::config()->isSetDisabled('system', 'debugging') ? 'disabled' : ''],
```
Comment (MrPetovan) [inline src/Module/Admin/Logs/Settings.php]: Please add a conditional help text or title attribute to explain why the value is disabled.

File: src/Module/Admin/Site.php [hunk 2]
```diff
@@ -413,15 +417,15 @@ protected function content(array $request = []): string
 			'$relocate_cmd'      => DI::l10n()->t('(Friendica directory)# bin/console relocate https://newdomain.com'),
 
 			// name, label, value, help string, extra data...
-			'$sitename'         => ['sitename', DI::l10n()->t('Site name'), DI::config()->get('config', 'sitename'), ''],
+			'$sitename'         => ['sitename', DI::l10n()->t('Site name'), DI::config()->get('config', 'sitename'), '', '', DI::config()->isSetDisabled('config', 'sitename') ? 'disabled' : ''],
```
Comment (MrPetovan) [inline src/Module/Admin/Site.php]: Please add a conditional help text or title attribute to explain why the value is disabled.

File: src/Module/Admin/Site.php [hunk 3]
```diff
@@ -413,15 +417,15 @@ protected function content(array $request = []): string
...
 			'$banner'           => ['banner', DI::l10n()->t('Banner/Logo'), $banner, ''],
 			'$email_banner'     => ['email_banner', DI::l10n()->t('Email Banner/Logo'), $email_banner, ''],
 			'$shortcut_icon'    => ['shortcut_icon', DI::l10n()->t('Shortcut icon'), DI::config()->get('system', 'shortcut_icon'), DI::l10n()->t('Link to an icon that will be used for browsers.')],
 			'$touch_icon'       => ['touch_icon', DI::l10n()->t('Touch icon'), DI::config()->get('system', 'touch_icon'), DI::l10n()->t('Link to an icon that will be used for tablets and mobiles.')],
 			'$additional_info'  => ['additional_info', DI::l10n()->t('Additional Info'), $additional_info, DI::l10n()->t('For public servers: you can add additional information here that will be listed at %s/servers.', Search::getGlobalDirectory())],
-			'$language'         => ['language', DI::l10n()->t('System language'), DI::config()->get('system', 'language'), '', $lang_choices],
+			'$language'         => ['language', DI::l10n()->t('System language'), DI::config()->get('system', 'language'), '', $lang_choices, DI::config()->isSetDisabled('system', 'language') ? 'disabled' : ''],
```
Comment (MrPetovan) [inline src/Module/Admin/Site.php]: Please add a conditional help text or title attribute to explain why the value is disabled.

File: src/Module/Admin/Storage.php [hunk 4]
```diff
@@ -129,6 +129,7 @@ protected function content(array $request = []): string
 				'prefix' => $storage_form_prefix,
 				'form'   => $storage_form,
 				'active' => $current_storage_backend instanceof ICanWriteToStorage && $name === $current_storage_backend::getName(),
+				'set_disabled' => DI::config()->isSetDisabled('storage', 'name'),
```
Comment (MrPetovan) [inline src/Module/Admin/Storage.php]: Please add a conditional help text or title attribute to explain why the value is disabled.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Admin/Storage.php (hunk 4 above)
Comment (nupplaphil) [reply src/Module/Admin/Storage.php]: I add a help text at the top of the setting, so I don't have to repeat myself for every storage

File: src/Core/Config/Model/DatabaseConfig.php [hunk 5]
```diff
@@ -83,7 +83,7 @@ public function get(string $cat, string $key = null, $default_value = null)
 	/** {@inheritDoc} */
 	public function isSetDisabled(string $cat, string $key): bool
 	{
-		return $this->cache->getSource($cat, $key) >= Cache::SOURCE_ENV;
+		return $this->cache->getSource($cat, $key) >= 0;
```
Comment (MrPetovan) [inline src/Core/Config/Model/DatabaseConfig.php]: Why did you remove the constant?

Comment (nupplaphil) [reply src/Core/Config/Model/DatabaseConfig.php]: Aaahhh debug leftover ... Thanks

File: src/Core/Config/Capability/IManageConfigValues.php [hunk 6]
```diff
@@ -58,6 +58,16 @@ public function reload();
...
 
+	/**
+	 * Returns true, if the current config cannot be changed
+	 *
+	 * @param string $cat The category of the configuration value
+	 * @param string $key The configuration key to query
+	 *
+	 * @return bool true, if set is disabled
+	 */
+	public function isSetDisabled(string $cat, string $key): bool;
```
Comment (MrPetovan) [inline src/Core/Config/Capability/IManageConfigValues.php]: Let's use a positive case instead, this will reduce the amount of boolean negation we have to do with `!`.
```suggestion
	public function isWritable(string $cat, string $key): bool;
```

File: src/Core/Config/Model/DatabaseConfig.php [hunk 7]
```diff
@@ -80,6 +80,12 @@ public function get(string $cat, string $key = null, $default_value = null)
 		return $this->cache->get($cat, $key) ?? $default_value;
 	}
 
+	/** {@inheritDoc} */
+	public function isSetDisabled(string $cat, string $key): bool
+	{
+		return $this->cache->getSource($cat, $key) >= Cache::SOURCE_ENV;
```
Comment (MrPetovan) [inline src/Core/Config/Model/DatabaseConfig.php]:
```suggestion
	public function isWritable(string $cat, string $key): bool
	{
		return $this->cache->getSource($cat, $key) < Cache::SOURCE_ENV;
```

File: src/Core/Config/Model/ReadOnlyFileConfig.php [hunk 8]
```diff
@@ -68,6 +68,12 @@ public function get(string $cat, string $key = null, $default_value = null)
 		return $this->configCache->get($cat, $key) ?? $default_value;
 	}
 
+	/** {@inheritDoc} */
+	public function isSetDisabled(string $cat, string $key): bool
+	{
+		return $this->configCache->getSource($cat, $key) >= Cache::SOURCE_ENV;
```
Comment (MrPetovan) [inline src/Core/Config/Model/ReadOnlyFileConfig.php]:
```suggestion
	public function isWritable(string $cat, string $key): bool
	{
		return $this->configCache->getSource($cat, $key) < Cache::SOURCE_ENV;
```

File: src/Module/Admin/Logs/Settings.php [hunk 9]
```diff
@@ -48,9 +48,15 @@ protected function post(array $request = [])
...
-		DI::config()->set('system', 'logfile', $logfile);
-		DI::config()->set('system', 'debugging', $debugging);
-		DI::config()->set('system', 'loglevel', $loglevel);
+		if (!DI::config()->isSetDisabled('system', 'logfile')) {
+			DI::config()->set('system', 'logfile', $logfile);
+		}
+		if (!DI::config()->isSetDisabled('system', 'debugging')) {
+			DI::config()->set('system', 'debugging', $debugging);
+		}
+		if (!DI::config()->isSetDisabled('system', 'loglevel')) {
```
Comment (MrPetovan) [inline src/Module/Admin/Logs/Settings.php]:
```suggestion
		if (DI::config()->isWritable('system', 'logfile')) {
			DI::config()->set('system', 'logfile', $logfile);
		}
		if (DI::config()->isWritable('system', 'debugging')) {
			DI::config()->set('system', 'debugging', $debugging);
		}
		if (DI::config()->isWritable('system', 'loglevel')) {
```

File: src/Module/Admin/Logs/Settings.php [hunk 10]
```diff
@@ -82,9 +88,9 @@ protected function content(array $request = []): string
...
-			'$loglevel' => ['loglevel', DI::l10n()->t("Log level"), DI::config()->get('system', 'loglevel'), "", $log_choices],
+			'$debugging' => ['debugging', DI::l10n()->t("Enable Debugging"), DI::config()->get('system', 'debugging'), DI::config()->isSetDisabled('system', 'debugging') ? DI::l10n()->t("<strong>Readonly</strong> because set by an environment variable") : '', DI::config()->isSetDisabled('system', 'debugging') ? 'disabled' : ''],
+			'$logfile' => ['logfile', DI::l10n()->t("Log file"), DI::config()->get('system', 'logfile'), DI::l10n()->t("Must be writable by web server. Relative to your Friendica top-level directory.") . (DI::config()->isSetDisabled('system', 'logfile') ? '<br>' . DI::l10n()->t("<strong>Readonly</strong> because set by an environment variable") : ''), "", DI::config()->isSetDisabled('system', 'logfile') ? 'disabled' : ''],
+			'$loglevel' => ['loglevel', DI::l10n()->t("Log level"), DI::config()->get('system', 'loglevel'), DI::config()->isSetDisabled('system', 'loglevel') ? DI::l10n()->t("<strong>Readonly</strong> because set by an environment variable") : '', $log_choices, DI::config()->isSetDisabled('system', 'loglevel') ? 'disabled' : ''],
```
Comment (MrPetovan) [inline src/Module/Admin/Logs/Settings.php]:
```suggestion
			'$debugging' => ['debugging', DI::l10n()->t('Enable Debugging'), DI::config()->get('system', 'debugging'), !DI::config()->isWritable('system', 'debugging') ? DI::l10n()->t('<strong>Read-only</strong> because it is set by an environment variable') : '', !DI::config()->isWritable('system', 'debugging') ? 'disabled' : ''],
			'$logfile' => ['logfile', DI::l10n()->t('Log file'), DI::config()->get('system', 'logfile'), DI::l10n()->t('Must be writable by web server. Relative to your Friendica top-level directory.') . (!DI::config()->isWritable('system', 'logfile') ? '<br>' . DI::l10n()->t('<strong>Read-only</strong> because it is set by an environment variable') : ''), '', !DI::config()->isWritable('system', 'logfile') ? 'disabled' : ''],
			'$loglevel' => ['loglevel', DI::l10n()->t("Log level"), DI::config()->get('system', 'loglevel'), !DI::config()->isWritable('system', 'loglevel') ? DI::l10n()->t('<strong>Read-only</strong> because it is set by an environment variable') : '', $log_choices, !DI::config()->isWritable('system', 'loglevel') ? 'disabled' : ''],
```

File: src/Module/Admin/Site.php [hunk 11]
```diff
@@ -165,7 +165,9 @@ protected function post(array $request = [])
 		$transactionConfig->set('system', 'poco_discovery'        , $poco_discovery);
 		$transactionConfig->set('system', 'poco_local_search'     , $poco_local_search);
 		$transactionConfig->set('system', 'nodeinfo'              , $nodeinfo);
-		$transactionConfig->set('config', 'sitename'              , $sitename);
+		if (!DI::config()->isSetDisabled('config', 'sitename')) {
+			$transactionConfig->set('config', 'sitename'          , $sitename);
```
Comment (MrPetovan) [inline src/Module/Admin/Site.php]:
```suggestion
		if (DI::config()->isWritable('config', 'sitename')) {
			$transactionConfig->set('config', 'sitename', $sitename);
```

File: src/Module/Admin/Site.php [hunk 12]
```diff
@@ -188,7 +190,9 @@ protected function post(array $request = [])
 		} else {
 			$transactionConfig->set('config', 'info', $additional_info);
 		}
-		$transactionConfig->set('system', 'language', $language);
+		if (!DI::config()->isSetDisabled('system', 'language')) {
```
Comment (MrPetovan) [inline src/Module/Admin/Site.php]:
```suggestion
		if (DI::config()->isWritable('system', 'language')) {
```

File: src/Module/Admin/Site.php [hunk 13]
```diff
@@ -413,15 +417,15 @@ protected function content(array $request = []): string
 			'$relocate_cmd'      => DI::l10n()->t('(Friendica directory)# bin/console relocate https://newdomain.com'),
 
 			// name, label, value, help string, extra data...
-			'$sitename'         => ['sitename', DI::l10n()->t('Site name'), DI::config()->get('config', 'sitename'), ''],
+			'$sitename'         => ['sitename', DI::l10n()->t('Site name'), DI::config()->get('config', 'sitename'), DI::config()->isSetDisabled('config', 'sitename') ? DI::l10n()->t("<strong>Readonly</strong> because set by an environment variable") : '', '', DI::config()->isSetDisabled('config', 'sitename') ? 'disabled' : ''],
```
Comment (MrPetovan) [inline src/Module/Admin/Site.php]:
```suggestion
			'$sitename'         => ['sitename', DI::l10n()->t('Site name'), DI::config()->get('config', 'sitename'), !DI::config()->isWritable('config', 'sitename') ? DI::l10n()->t('<strong>Read-only</strong> because it is set by an environment variable') : '', '', !DI::config()->isWritable('config', 'sitename') ? 'disabled' : ''],
```

File: src/Module/Admin/Site.php [hunk 14]
```diff
@@ -413,15 +417,15 @@ protected function content(array $request = []): string
...
 			'$email_banner'     => ['email_banner', DI::l10n()->t('Email Banner/Logo'), $email_banner, ''],
 			'$shortcut_icon'    => ['shortcut_icon', DI::l10n()->t('Shortcut icon'), DI::config()->get('system', 'shortcut_icon'), DI::l10n()->t('Link to an icon that will be used for browsers.')],
 			'$touch_icon'       => ['touch_icon', DI::l10n()->t('Touch icon'), DI::config()->get('system', 'touch_icon'), DI::l10n()->t('Link to an icon that will be used for tablets and mobiles.')],
 			'$additional_info'  => ['additional_info', DI::l10n()->t('Additional Info'), $additional_info, DI::l10n()->t('For public servers: you can add additional information here that will be listed at %s/servers.', Search::getGlobalDirectory())],
-			'$language'         => ['language', DI::l10n()->t('System language'), DI::config()->get('system', 'language'), '', $lang_choices],
+			'$language'         => ['language', DI::l10n()->t('System language'), DI::config()->get('system', 'language'), DI::config()->isSetDisabled('system', 'language') ? DI::l10n()->t("<strong>Readonly</strong> because set by an environment variable") : '', $lang_choices, DI::config()->isSetDisabled('system', 'language') ? 'disabled' : ''],
```
Comment (MrPetovan) [inline src/Module/Admin/Site.php]:
```suggestion
			'$language'         => ['language', DI::l10n()->t('System language'), DI::config()->get('system', 'language'), !DI::config()->isWritable('system', 'language') ? DI::l10n()->t('<strong>Read-only</strong> because it is set by an environment variable") : '', $lang_choices, !DI::config()->isWritable('system', 'language') ? 'disabled' : ''],
```

File: src/Module/Admin/Storage.php [hunk 15]
```diff
@@ -76,7 +76,7 @@ protected function post(array $request = [])
 			}
 		}
 
-		if (!empty($_POST['submit_save_set'])) {
+		if (!empty($_POST['submit_save_set']) && !DI::config()->isSetDisabled('storage', 'name') ) {
```
Comment (MrPetovan) [inline src/Module/Admin/Storage.php]:
```suggestion
		if (!empty($_POST['submit_save_set']) && DI::config()->isWritable('storage', 'name') ) {
```

File: src/Module/Admin/Storage.php (hunk 4 above)
Comment (MrPetovan) [inline src/Module/Admin/Storage.php]: This should be set only once in the base template variable array lines 139-151, not in each storage backend array.
```suggestion
				'is_writable' => DI::config()->isWritable('storage', 'name'),
```

File: src/Module/Admin/Storage.php [hunk 16]
```diff
@@ -145,6 +146,7 @@ protected function content(array $request = []): string
 			'$save_reload'           => DI::l10n()->t('Save & Reload'),
 			'$noconfig'              => DI::l10n()->t('This backend doesn\'t have custom settings'),
 			'$form_security_token'   => self::getFormSecurityToken("admin_storage"),
+			'$storagebackend_ro_txt' => DI::config()->isSetDisabled('storage', 'name') ? DI::l10n()->t("Changing the current backend is prohibited because it is set by an environment variable") : '',
```
Comment (MrPetovan) [inline src/Module/Admin/Storage.php]:
```suggestion
			'$storagebackend_ro_txt' => !DI::config()->isWritable('storage', 'name') ? DI::l10n()->t('Changing the current backend is prohibited because it is set by an environment variable') : '',
```

File: tests/src/Core/Config/ConfigTest.php [hunk 17]
```diff
@@ -566,4 +566,74 @@ public function testSerializedValues($value, $assertion)
...
+		$configFileManager = new ConfigFileManager($this->root->url(), $this->root->url() . '/config/', $this->root->url() . '/static/', $server);
+		$configFileManager->setupCache($this->configCache);
+		$config = new DatabaseConfig($this->getDbInstance(), $this->configCache);
+
+		foreach ($data as $category => $keyvalues) {
+			foreach ($keyvalues as $key => $value) {
+				if (!empty($assertDisabled[$category][$key])) {
+					static::assertTrue($config->isSetDisabled($category, $key), sprintf("%s.%s is not true", $category, $key));
+				} else {
+					static::assertFalse($config->isSetDisabled($category, $key), sprintf("%s.%s is not false", $category, $key));
```
Comment (MrPetovan) [inline tests/src/Core/Config/ConfigTest.php]:
```suggestion
					static::assertTrue($config->isWritable($category, $key), sprintf('%s.%s is not true', $category, $key));
				} else {
					static::assertFalse($config->isWritable($category, $key), sprintf('%s.%s is not false', $category, $key));
```

File: view/templates/admin/storage.tpl [hunk 18]
```diff
@@ -19,12 +20,14 @@
 
 		{{if $storage.form}}
 		<input type="submit" name="submit_save" value="{{$save}}"/>
-		{{if $storage.active}}
+        {{if ! $storage.set_disabled}}
```
Comment (MrPetovan) [inline view/templates/admin/storage.tpl]:
```suggestion
        {{if $is_writable}}
```

File: view/templates/admin/storage.tpl [hunk 19]
```diff
@@ -19,12 +20,14 @@
...
+				{{if $storage.active}}
 		<input type="submit" name="submit_save_set" value="{{$save_reload}}"/>
-		{{else}}
+				{{else}}
 		<input type="submit" name="submit_save_set" value="{{$save_use}}"/>
+				{{/if}}
 		{{/if}}
-		{{else}}
+		{{elseif ! $storage.set_disabled}}
```
Comment (MrPetovan) [inline view/templates/admin/storage.tpl]:
```suggestion
		{{elseif $is_writable}}
```

File: view/theme/frio/templates/admin/storage.tpl [hunk 20]
```diff
@@ -5,6 +5,9 @@
 
 	<div class="well well-lg">
         	{{$label_current}}: <b>{{$storagebackend}}</b>
+			{{if $storagebackend_ro_txt}}
+			<p><p><i>{{$storagebackend_ro_txt nofilter}}</i>
```
Comment (MrPetovan) [inline view/theme/frio/templates/admin/storage.tpl]:
```suggestion
			<br><i>{{$storagebackend_ro_txt nofilter}}</i>
```

File: view/theme/frio/templates/admin/storage.tpl [hunk 21]
```diff
@@ -33,12 +36,14 @@
 				<div class="panel-footer">
 					{{if $storage.form}}
 					<input type="submit" name="submit_save" class="btn btn-primary" value="{{$save}}"/>
-						{{if $storage.active}}
+	                    {{if ! $storage.set_disabled}}
```
Comment (MrPetovan) [inline view/theme/frio/templates/admin/storage.tpl]:
```suggestion
	                    {{if $is_writable}}
```

File: view/theme/frio/templates/admin/storage.tpl [hunk 22]
```diff
@@ -33,12 +36,14 @@
...
+							{{if $storage.active}}
 					<input type="submit" name="submit_save_set" class="btn btn-primary" value="{{$save_reload}}"/>
-						{{else}}
+							{{else}}
 					<input type="submit" name="submit_save_set" class="btn btn-primary" value="{{$save_use}}"/>
+							{{/if}}
 						{{/if}}
-					{{else}}
+					{{elseif ! $storage.set_disabled }}
```
Comment (MrPetovan) [inline view/theme/frio/templates/admin/storage.tpl]:
```suggestion
					{{elseif $is_writable}}
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Admin/Storage.php [hunk 23]
```diff
@@ -145,6 +146,7 @@ protected function content(array $request = []): string
 			'$save_reload'           => DI::l10n()->t('Save & Reload'),
 			'$noconfig'              => DI::l10n()->t('This backend doesn\'t have custom settings'),
 			'$form_security_token'   => self::getFormSecurityToken("admin_storage"),
+			'$storagebackend_ro_txt' => !DI::config()->isWritable('storage', 'name') ? DI::l10n()->t('Changing the current backend is prohibited because it is set by an environment variable') : '',
```
Comment (MrPetovan) [inline src/Module/Admin/Storage.php:148]: You need to do this for the `$is_writable` variable to be set in the templates.
```suggestion
			'$storagebackend_ro_txt' => !DI::config()->isWritable('storage', 'name') ? DI::l10n()->t('Changing the current backend is prohibited because it is set by an environment variable') : '',
			'$is_writable' => DI::config()->isWritable('storage', 'name'),
```

File: src/Module/Admin/Storage.php [hunk 24]
```diff
@@ -129,6 +129,7 @@ protected function content(array $request = []): string
 				'prefix' => $storage_form_prefix,
 				'form'   => $storage_form,
 				'active' => $current_storage_backend instanceof ICanWriteToStorage && $name === $current_storage_backend::getName(),
+				'is_writable' => DI::config()->isWritable('storage', 'name'),
```
Comment (MrPetovan) [inline src/Module/Admin/Storage.php]:
```suggestion
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Admin/Site.php [hunk 25]
```diff
@@ -425,7 +425,7 @@ protected function content(array $request = []): string
...
 			'$touch_icon'       => ['touch_icon', DI::l10n()->t('Touch icon'), DI::config()->get('system', 'touch_icon'), DI::l10n()->t('Link to an icon that will be used for tablets and mobiles.')],
 			'$additional_info'  => ['additional_info', DI::l10n()->t('Additional Info'), $additional_info, DI::l10n()->t('For public servers: you can add additional information here that will be listed at %s/servers.', Search::getGlobalDirectory())],
-			'$language'         => ['language', DI::l10n()->t('System language'), DI::config()->get('system', 'language'), !DI::config()->isWritable('system', 'language') ? DI::l10n()->t('<strong>Read-only</strong> because it is set by an environment variable") : '', $lang_choices, !DI::config()->isWritable('system', 'language') ? 'disabled' : ''],
+			'$language'         => ['language', DI::l10n()->t('System language'), DI::config()->get('system', 'language'), !DI::config()->isWritable('system', 'language') ? DI::l10n()->t("<strong>Read-only</strong> because it is set by an environment variable") : '', $lang_choices, !DI::config()->isWritable('system', 'language') ? 'disabled' : ''],
```
Comment (MrPetovan) [inline src/Module/Admin/Site.php:428]: There's no need for double quotes here (no literal single quotes, no variable interpolation. I'm sorry I didn't pair the single quotes in my GitHub review. 😛
```suggestion
			'$language'         => ['language', DI::l10n()->t('System language'), DI::config()->get('system', 'language'), !DI::config()->isWritable('system', 'language') ? DI::l10n()->t('<strong>Read-only</strong> because it is set by an environment variable') : '', $lang_choices, !DI::config()->isWritable('system', 'language') ? 'disabled' : ''],
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
Huh, a quote change trips the messages.po check.

**Comment by nupplaphil** [comment]
Yes because the wrong quote accidentally deleted a lot of strings in the messages.po -.-**

I'll fix it tomorrow ..
